pkg/cluster/ClusterNoteService.go

```diff
@@ -0,0 +1,208 @@
+		}
+	}
+	bean.Id = model.Id
+	// audit the existing description to cluster audit history
+	clusterAudit := &repository.ClusterNoteHistory{
+		NoteId:      bean.Id,
+		Description: bean.Description,
+	}
+	clusterAudit.CreatedBy = userId
+	clusterAudit.UpdatedBy = userId
+	clusterAudit.CreatedOn = time.Now()
+	clusterAudit.UpdatedOn = time.Now()
+	err = impl.clusterNoteHistoryRepository.SaveHistory(clusterAudit)
+	if err != nil {
+		return nil, err
+	}
+	return bean, err
+}
+
+func (impl *ClusterNoteServiceImpl) FindByClusterId(id int) (*ClusterNoteBean, error) {
+	model, err := impl.clusterNoteRepository.FindByClusterId(id)
+	if err != nil {
+		return nil, err
+	}
+	bean := &ClusterNoteBean{
+		Id:          model.Id,
+		ClusterId:   model.ClusterId,
+		Description: model.Description,
+		CreatedBy:   int(model.CreatedBy),
+		CreatedOn:   model.CreatedOn,
+	}
+	return bean, nil
+}
+
+func (impl *ClusterNoteServiceImpl) FindByClusterIds(ids []int) ([]ClusterNoteBean, error) {
+	models, err := impl.clusterNoteRepository.FindByClusterIds(ids)
+	if err != nil {
+		return nil, err
+	}
+	var beans []ClusterNoteBean
+
+	for _, model := range models {
+		beans = append(beans, ClusterNoteBean{
+			Id:          model.Id,
+			ClusterId:   model.ClusterId,
+			Description: model.Description,
+			CreatedBy:   int(model.CreatedBy),
+			CreatedOn:   model.CreatedOn,
+		})
+	}
+	return beans, nil
+}
+
+func (impl *ClusterNoteServiceImpl) Update(bean *ClusterNoteBean, userId int32) (*ClusterNoteBean, error) {
+	model, err := impl.clusterNoteRepository.FindByClusterId(bean.ClusterId)
+	if err != nil {
+		impl.logger.Error(err)
+		return nil, err
+	}
+	if model.Id == 0 {
+		impl.logger.Errorw("error on fetching cluster note, not found", "id", bean.Id)
+		return nil, fmt.Errorf("cluster note not found")
+	}
+	if bean.Description == "" {
+		// blank description
+		impl.logger.Errorw("blank cluster note, blank", "id", bean.Id)
+		return nil, fmt.Errorf("blank cluster note")
+	}
+	if bean.Description == "" || bean.Description == model.Description {
+		// no change in description
+		impl.logger.Errorw("no change found in cluster note, duplicate", "id", bean.Id)
+		return nil, fmt.Errorf("duplicate cluster note")
+	}
+	// update the cluster description with new data
+	model.Description = bean.Description
+	model.UpdatedBy = userId
+	model.UpdatedOn = time.Now()
+
+	err = impl.clusterNoteRepository.Update(model)
+	if err != nil {
+		err = &util.ApiError{
+			Code:            constants.ClusterUpdateDBFailed,
+			InternalMessage: "cluster note update failed in db",
+			UserMessage:     fmt.Sprintf("requested by %d", userId),
+		}
+		return bean, err
+	}
+	bean.Id = model.Id
+	// audit the existing description to cluster audit history
+	clusterAudit := &repository.ClusterNoteHistory{
+		NoteId:      bean.Id,
+		Description: bean.Description,
+	}
+	clusterAudit.CreatedBy = userId
+	clusterAudit.UpdatedBy = userId
+	clusterAudit.CreatedOn = time.Now()
```

Comment:
on update why created by and created on ?

Reply:
created by and created on is required for saving the record to the audit history